deepchem/models/tensorgraph/tensor_graph.py

```diff
@@ -170,35 +226,63 @@ def predict_on_batch(self, X, sess=None):
     # Returns
         A Numpy array of predictions.
     """
-    retval = self.predict_proba_on_batch(X, sess)
+    retval = self.predict_proba_on_generator(generator, sess)
     if self.mode == 'classification':
-      return from_one_hot(retval, axis=2)
+      retval = np.expand_dims(from_one_hot(retval, axis=2), axis=1)
     return retval
 
-  def predict_proba_on_batch(self, X, sess=None):
-
-    def predict():
-      out_tensors = [x.out_tensor for x in self.outputs]
-      fetches = out_tensors
-      feed_dict = self._construct_feed_dict(X, None, None, None)
-      fetched_values = sess.run(fetches, feed_dict=feed_dict)
-      return np.array(fetched_values)
+  def predict_proba_on_generator(self, generator, sess=None):
+    """
+    TODO: Do transformers even make sense here?
 
+    Returns:
+      y_pred: numpy ndarray of shape (n_samples, n_classes*n_tasks)
+    """
     if not self.built:
       self.build()
-    if sess is None:
-      saver = tf.train.Saver()
+
+    def predict_closure(session):
```

Comment:
I don't need it anymore.  It's a remnant that was needed for a single entry point which session was activated and the context.